### ISSUE TITLE
loss normalization 

### ISSUE THREAD

**Issue by jiangxiluning** [bentrevett/pytorch-seq2seq#23]
why do we need to normalize loss by number of tokens?

**Comment by bentrevett** [comment]
I mainly did it because most other open source seq2seq implementations in PyTorch averaged over the number of tokens. This is also the default method of handling sequence to sequence losses in Tensorflow: https://www.tensorflow.org/api_docs/python/tf/contrib/seq2seq/sequence_loss

Feel free to try summing the losses across time-steps and letting me know if it improves results!
